corehq/apps/repeaters/models.py

```diff
@@ -301,7 +283,15 @@ def get_auth(self):
     def fire_for_record(self, repeat_record):
         headers = self.get_headers(repeat_record)
         auth = self.get_auth()
-        payload = self.get_payload_and_handle_exception(repeat_record)
+
+        try:
+            payload = self.get_payload(repeat_record)
+        except Exception as e:
+            # todo: seems like this just does everything that handle_exception does but not as well.
+            # todo:   seems like they could be combined
```

Comment:
Yeah, I think previously the `handle_payload_exception` was also calling `raise`. It might not have gotten updated when we added the `cancelled` state.